modules/form-completion-status/src/main/java/ca/sickkids/ccm/lfs/formcompletionstatus/AnswerCompletionStatusEditor.java

```diff
@@ -0,0 +1,604 @@
+            int numAnswers = iterableLength(nodeAnswers);
+            ArrayList<String> statusFlags = new ArrayList<String>();
+            if (checkInvalidAnswer(questionNode, numAnswers)) {
+                statusFlags.add(STATUS_FLAG_INVALID);
+                statusFlags.add(STATUS_FLAG_INCOMPLETE);
+            } else {
+                /*
+                 * We are here because:
+                 *     - minAnswers == 0 && maxAnswers == 0
+                 *     - minAnswers == 0 && maxAnswers == 1 && numAnswers in range [0,1] (eg. optional radio button)
+                 *     - minAnswers == 1 && maxAnswers == 0 && numAnswers in range [1,+INF) (eg. mandatory checkboxes)
+                 *     - minAnswers == 1 && maxAnswers == 1 && numAnswers == 1 (eg. mandatory radio button)
+                 *     - minAnswers == N && maxAnswers == 0 && numAnswers in range [N,+INF)
+                 *         (eg. at least N (inclusive) checkboxes must be selected)
+                 *     - minAnswers == 0 && maxAnswers == M && numAnswers in range [0, M]
+                 *         (eg. at most M (inclusive) checkboxes must be selected)
+                 *     - minAnswers == N && maxAnswers == M && numAnswers in range [N,M]
+                 *         (eg. between N (inclusive) and M (inclusive) checkboxes must be selected)
+                 */
+
+                /*
+                 * TODO: Implement validation rules and check them here
+                 * Remove INVALID and INCOMPLETE flags if all validation rules pass
+                 */
+            }
+            this.currentNodeBuilder.setProperty(STATUS_FLAGS, statusFlags, Type.STRINGS);
+            //Summarize all parents
+            try {
+                summarizeBuilders(this.currentNodeBuilderPath);
+            } catch (RepositoryException e) {
+                LOGGER.warn("Could not run summarize()");
+            }
+        }
+    }
+
+    // Called when a property is deleted
+    @Override
+    public void propertyDeleted(PropertyState before)
+        throws CommitFailedException
+    {
+        Node questionNode = getQuestionNode(this.currentNodeBuilder);
+        if (questionNode != null) {
+            if (PROP_VALUE.equals(before.getName())) {
+                ArrayList<String> statusFlags = new ArrayList<String>();
+                //Only add the INVALID,INCOMPLETE flags if the given question requires more than zero answers
+                if (checkInvalidAnswer(questionNode, 0)) {
+                    statusFlags.add(STATUS_FLAG_INVALID);
+                    statusFlags.add(STATUS_FLAG_INCOMPLETE);
+                }
+                this.currentNodeBuilder.setProperty(STATUS_FLAGS, statusFlags, Type.STRINGS);
+
+                //Summarize all parents
+                try {
+                    summarizeBuilders(this.currentNodeBuilderPath);
+                } catch (RepositoryException e) {
+                    LOGGER.warn("Could not run summarize()");
```

Comment:
```suggestion
                    LOGGER.warn("Could not run summarizeBuilders()");
                    LOGGER.warn(e.toString());
```
It might also be worth printing the stack trace while we're at it.

Reply:
Done :heavy_check_mark: